### PR TITLE
Ndev 3212 tests for block number

### PR DESCRIPTION
**Pull request by andrei-neon**

### CODE REVIEW

File: contracts/common/Block.sol [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+contract BlockTimestamp {
+
+    function getBlockTimestamp() public view returns (uint256) {
+        return block.timestamp;
+    }
+
+}
+
+contract BlockTimestampCaller {
```
Comment (kristinaNikolaevaa) [inline contracts/common/Block.sol]: you can merge all 3 contracts for BlockTimestamp in 1 
and the same for block number to simplify fixutres

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.getBlockTimestamp().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        assert hex(contract.functions.getBlockTimestamp().call()) == validate_response.result.timestamp
+
+    def test_block_timestamp_event(self, block_timestamp_contracts, json_rpc_client):
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: test_block_timestamp_send_simple_trx

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        assert hex(event_logs[0]["args"]["block_timestamp"]) == validate_response.result.timestamp
+
+    def test_block_timestamp_iterative(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[1]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.call_events_in_loop().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: maybe it's better to rename variables
it is not good to have response and resp in 1 space

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        sender_account = self.accounts[0]
+
+        text = "".join([random.choice(string.ascii_uppercase) for _ in range(5)])
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.addData(text, 1).build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        event_logs = contract.events.DataAdded().process_receipt(resp)
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: I think we don't need this test 
test_block_timestamp_mapping_get_data - almost the same but better

File: integration/tests/conftest.py [hunk 5]
```diff
@@ -573,3 +573,21 @@ def counter_resource_address(call_solana_caller, accounts, web3_client):
...
+def block_timestamp_contracts(web3_client, accounts):
+    contract_a, _ = web3_client.deploy_and_get_contract("common/Block.sol", "0.8.10", accounts[0], contract_name="BlockTimestamp")
+    contract_b, _ = web3_client.deploy_and_get_contract("common/Block.sol", "0.8.10", accounts[0], contract_name="BlockTimestampCaller")
+    contract_c, receipt = web3_client.deploy_and_get_contract("common/Block.sol", "0.8.10", accounts[0], contract_name="BlockTimestampInitializer")
+    contract_d, _ = web3_client.deploy_and_get_contract("common/Block.sol", "0.8.10", accounts[0], contract_name="BlockTimestampMapping")
+    return contract_a, contract_b, (contract_c, receipt), contract_d
+
+
+@pytest.fixture(scope="class")
+def block_number_contracts(web3_client, accounts):
```
Comment (kristinaNikolaevaa) [inline integration/tests/conftest.py]: please move fixtures to class where it's used if they are needed only in 1 place

**Review by kristinaNikolaevaa — CHANGES_REQUESTED** [review]

**Comment by kristinaNikolaevaa** [comment]
dont forget about md files in docs

File: contracts/common/Block.sol (hunk 1 above)
Comment (andrei-neon) [reply contracts/common/Block.sol]: Fixed ✅

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py (hunk 2 above)
Comment (andrei-neon) [reply integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: Fixed ✅

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py (hunk 3 above)
Comment (andrei-neon) [reply integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: Fixed ✅

File: integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py (hunk 4 above)
Comment (andrei-neon) [reply integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py]: Fixed ✅

File: integration/tests/conftest.py (hunk 5 above)
Comment (andrei-neon) [reply integration/tests/conftest.py]: Fixed ✅

**Pull request closed without merging**